src/blueapi/worker/reworker.py

```diff
@@ -64,20 +67,33 @@ def __init__(
         self._data_events = EventPublisher()
         self._status_lock = RLock()
         self._status_snapshot = {}
+        self._request_exit = Event()
+        self._finished = Event()
 
     def submit_task(self, name: str, task: Task) -> None:
         active_task = ActiveTask(name, task)
         LOGGER.info(f"Submitting: {active_task}")
         self._task_queue.put(active_task)
 
-    def run_forever(self) -> None:
+    def start(self) -> None:
+        LOGGER.info("Starting worker in own thread")
+        run_worker_in_own_thread(self)
+
+    def stop(self) -> None:
+        LOGGER.info("Stopping worker thread")
+        self._request_exit.set()
+        self._finished.wait()
+        LOGGER.info("Worker finished")
```

Comment:
Should clear the status of these events after stopping, in case we later call start() again (or implement a restart)